modules/home-manager/neovim.nix

```
@@ -1,16 +1,22 @@
 { catppuccinLib }:
-{
-  config,
-  lib,
-  ...
-}:
-
+{ config, lib, ... }:
 let
   cfg = config.catppuccin.nvim;
-in
+  pluginSettings = lib.generators.toLua { } (defaultConfig // cfg.extraConfig);
 
+  defaultConfig = {
+    compile_path = lib.generators.mkLuaInline "compile_path";
+    flavour = cfg.flavor;
+  };
+in
 {
-  options.catppuccin.nvim = catppuccinLib.mkCatppuccinOption { name = "neovim"; };
+  options.catppuccin.nvim = catppuccinLib.mkCatppuccinOption { name = "neovim"; } // {
+    extraConfig = lib.mkOption {
+      description = "Extra settings that will be passed to the setup function.";
+      default = { };
+      type = lib.types.attrs;
```

```suggestion
      type = lib.types.attrs;
      apply = lib.recursiveUpdate defaultConfig;
```
It'd be better to apply this to the option itself so that the actual final value of it can be easily evaluated (as opposed to currently where the final configuration used is limited to the above `let ...; in` scope)